[
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "I'm taking a look at this.",
      "created_at" : "2023-11-29T13:14:38Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1831875763",
      "id" : 1831875763,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tMDSz",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1831875763/reactions"
      },
      "updated_at" : "2023-11-29T13:14:38Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1831875763",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/19157360?v=4",
         "events_url" : "https://api.github.com/users/0xB10C/events{/privacy}",
         "followers_url" : "https://api.github.com/users/0xB10C/followers",
         "following_url" : "https://api.github.com/users/0xB10C/following{/other_user}",
         "gists_url" : "https://api.github.com/users/0xB10C/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/0xB10C",
         "id" : 19157360,
         "login" : "0xB10C",
         "node_id" : "MDQ6VXNlcjE5MTU3MzYw",
         "organizations_url" : "https://api.github.com/users/0xB10C/orgs",
         "received_events_url" : "https://api.github.com/users/0xB10C/received_events",
         "repos_url" : "https://api.github.com/users/0xB10C/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/0xB10C/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/0xB10C/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/0xB10C"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "```\r\nnode1 [httpworker.2] [rpc/request.cpp:181] [parse] [rpc] ThreadRPCServer method=addpeeraddress user=__cookie__ \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 2, tried 1, total 3 started \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: completed (0.60ms) \r\nnode1 [httpworker.2] [addrman.cpp:615] [AddSingle] [addrman] Added nrfj6inpyf73gpkyool35hcmne5zwfmse3jl3aw23vk7chdemalyaqad.onion:0 to new[238][20] \r\nnode1 [httpworker.2] [addrman.cpp:687] [Add_] [addrman] Added 1 addresses (of 1) from nrfj6inpyf73gpkyool35hcmne5zwfmse3jl3aw23vk7chdemalyaqad.onion: 1 tried, 3 new \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: completed (0.87ms) \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: completed (0.87ms) \r\nnode1 [httpworker.2] [addrman.cpp:668] [Good_] [addrman] Collision with 1.2.3.4:8333 while attempting to move nrfj6inpyf73gpkyool35hcmne5zwfmse3jl3aw23vk7chdemalyaqad.onion:0 to tried table. Collisions=1 \r\nnode1 [httpworker.2] [logging/timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started \r\ntest TestFramework (DEBUG): Added nrfj6inpyf73gpkyool35hcmne5zwfmse3jl3aw23vk7chdemalyaqad.onion to tried table \r\ntest TestFramework (DEBUG): Test that the newly added addresses appear in getrawaddrman \r\n```\r\n\r\nseems like this didn't work:\r\n\r\nhttps://github.com/bitcoin/bitcoin/blob/8cf2137dbeaa681f0178f1e2fd70b639e6d4f717/test/functional/rpc_net.py#L492-L504\r\n\r\nDid `addpeeraddress` return `{\"success\": true}` even if there was a collision while moving to tried? \r\n\r\n(still investigating)",
      "created_at" : "2023-11-29T16:36:25Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1832296894",
      "id" : 1832296894,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tNqG-",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832296894/reactions"
      },
      "updated_at" : "2023-11-29T16:36:25Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832296894",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/19157360?v=4",
         "events_url" : "https://api.github.com/users/0xB10C/events{/privacy}",
         "followers_url" : "https://api.github.com/users/0xB10C/followers",
         "following_url" : "https://api.github.com/users/0xB10C/following{/other_user}",
         "gists_url" : "https://api.github.com/users/0xB10C/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/0xB10C",
         "id" : 19157360,
         "login" : "0xB10C",
         "node_id" : "MDQ6VXNlcjE5MTU3MzYw",
         "organizations_url" : "https://api.github.com/users/0xB10C/orgs",
         "received_events_url" : "https://api.github.com/users/0xB10C/received_events",
         "repos_url" : "https://api.github.com/users/0xB10C/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/0xB10C/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/0xB10C/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/0xB10C"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "I think that the problem is a rare collision in the tried table: \r\n` node1 2023-11-28T23:28:31.447418Z (mocktime: 2023-11-28T23:28:28Z) [httpworker.2] [addrman.cpp:668] [Good_] [addrman] Collision with 1.2.3.4:8333 while attempting to move nrfj6inpyf73gpkyool35hcmne5zwfmse3jl3aw23vk7chdemalyaqad.onion:0 to tried table. Collisions=1 `:\r\n\r\n`addpeeraddress` reports success if the first part (adding to new) succeeded, but the second part (moving to tried) fails due to a collision (so that the addr stays in new).",
      "created_at" : "2023-11-29T16:57:59Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1832335914",
      "id" : 1832335914,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tNzoq",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832335914/reactions"
      },
      "updated_at" : "2023-11-29T16:57:59Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832335914",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/48763452?v=4",
         "events_url" : "https://api.github.com/users/mzumsande/events{/privacy}",
         "followers_url" : "https://api.github.com/users/mzumsande/followers",
         "following_url" : "https://api.github.com/users/mzumsande/following{/other_user}",
         "gists_url" : "https://api.github.com/users/mzumsande/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/mzumsande",
         "id" : 48763452,
         "login" : "mzumsande",
         "node_id" : "MDQ6VXNlcjQ4NzYzNDUy",
         "organizations_url" : "https://api.github.com/users/mzumsande/orgs",
         "received_events_url" : "https://api.github.com/users/mzumsande/received_events",
         "repos_url" : "https://api.github.com/users/mzumsande/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/mzumsande/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/mzumsande/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/mzumsande"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "@mzumsande is there a reason why `addpeeraddress` should return true if it fails to put in in the table we asked it to do? Sounds like a bug to me. Or, it should at least be documented clearly.\r\n\r\nhttps://github.com/bitcoin/bitcoin/blob/7bc8c5312bf51edd3d85e1599b5d6ae32c8fd7b5/src/rpc/net.cpp#L981-L987",
      "created_at" : "2023-11-29T18:03:45Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1832443461",
      "id" : 1832443461,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tON5F",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832443461/reactions"
      },
      "updated_at" : "2023-11-29T18:03:45Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832443461",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/19157360?v=4",
         "events_url" : "https://api.github.com/users/0xB10C/events{/privacy}",
         "followers_url" : "https://api.github.com/users/0xB10C/followers",
         "following_url" : "https://api.github.com/users/0xB10C/following{/other_user}",
         "gists_url" : "https://api.github.com/users/0xB10C/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/0xB10C",
         "id" : 19157360,
         "login" : "0xB10C",
         "node_id" : "MDQ6VXNlcjE5MTU3MzYw",
         "organizations_url" : "https://api.github.com/users/0xB10C/orgs",
         "received_events_url" : "https://api.github.com/users/0xB10C/received_events",
         "repos_url" : "https://api.github.com/users/0xB10C/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/0xB10C/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/0xB10C/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/0xB10C"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "I don't think there is a reason, but returning `false` wouldn't be great either. It might be best could return an error code with 3 values (not added to new; added to new but failed to move to tried; everything successful).",
      "created_at" : "2023-11-29T18:08:48Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1832450581",
      "id" : 1832450581,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tOPoV",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832450581/reactions"
      },
      "updated_at" : "2023-11-29T18:09:57Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1832450581",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/48763452?v=4",
         "events_url" : "https://api.github.com/users/mzumsande/events{/privacy}",
         "followers_url" : "https://api.github.com/users/mzumsande/followers",
         "following_url" : "https://api.github.com/users/mzumsande/following{/other_user}",
         "gists_url" : "https://api.github.com/users/mzumsande/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/mzumsande",
         "id" : 48763452,
         "login" : "mzumsande",
         "node_id" : "MDQ6VXNlcjQ4NzYzNDUy",
         "organizations_url" : "https://api.github.com/users/mzumsande/orgs",
         "received_events_url" : "https://api.github.com/users/mzumsande/received_events",
         "repos_url" : "https://api.github.com/users/mzumsande/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/mzumsande/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/mzumsande/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/mzumsande"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "Improving `addpeeraddress tried=true` failure behavior in #28998 ",
      "created_at" : "2023-12-05T14:23:30Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1840892172",
      "id" : 1840892172,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII585tuckM",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1840892172/reactions"
      },
      "updated_at" : "2023-12-05T14:23:30Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1840892172",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/19157360?v=4",
         "events_url" : "https://api.github.com/users/0xB10C/events{/privacy}",
         "followers_url" : "https://api.github.com/users/0xB10C/followers",
         "following_url" : "https://api.github.com/users/0xB10C/following{/other_user}",
         "gists_url" : "https://api.github.com/users/0xB10C/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/0xB10C",
         "id" : 19157360,
         "login" : "0xB10C",
         "node_id" : "MDQ6VXNlcjE5MTU3MzYw",
         "organizations_url" : "https://api.github.com/users/0xB10C/orgs",
         "received_events_url" : "https://api.github.com/users/0xB10C/received_events",
         "repos_url" : "https://api.github.com/users/0xB10C/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/0xB10C/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/0xB10C/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/0xB10C"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "Same or similar issue?\r\n\r\n```shell\r\n test  2024-02-28T10:08:07.041000Z TestFramework (ERROR): Assertion failed \r\n                                   Traceback (most recent call last):\r\n                                     File \"/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py\", line 131, in main\r\n                                       self.run_test()\r\n                                     File \"/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/rpc_net.py\", line 63, in run_test\r\n                                       self.test_getpeerinfo()\r\n                                     File \"/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/rpc_net.py\", line 123, in test_getpeerinfo\r\n                                       assert_equal(\r\n                                     File \"/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py\", line 57, in assert_equal\r\n                                       raise AssertionError(\"not(%s)\" % \" == \".join(str(arg) for arg in (thing1, thing2) + args))\r\n                                   AssertionError: not({'id': 2, 'network': 'not_publicly_routable', 'services': '0000000000000000', 'servicesnames': [], 'relaytxes': False, 'lastsend': 1709114886, 'lastrecv': 1709114886, 'last_transaction': 0, 'last_block': 0, 'conntime': 1709114886, 'timeoffset': 0, 'version': 0, 'subver': '', 'inbound': True, 'bip152_hb_to': False, 'bip152_hb_from': False, 'startingheight': -1, 'presynced_headers': -1, 'synced_headers': -1, 'synced_blocks': -1, 'inflight': [], 'addr_relay_enabled': False, 'addr_processed': 0, 'addr_rate_limited': 0, 'permissions': [], 'minfeefilter': Decimal('0E-8'), 'bytessent_per_msg': {}, 'bytesrecv_per_msg': {}, 'connection_type': 'inbound', 'transport_protocol_type': 'detecting', 'session_id': ''} == {'addr_processed': 0, 'addr_rate_limited': 0, 'addr_relay_enabled': False, 'bip152_hb_from': False, 'bip152_hb_to': False, 'bytesrecv_per_msg': {}, 'bytessent_per_msg': {}, 'connection_type': 'inbound', 'conntime': 1709114886, 'id': 2, 'inbound': True, 'inflight': [], 'last_block': 0, 'last_transaction': 0, 'lastrecv': 1709114886, 'lastsend': 1709114886, 'minfeefilter': Decimal('0E-8'), 'network': 'not_publicly_routable', 'permissions': [], 'presynced_headers': -1, 'relaytxes': False, 'services': '0000000000000000', 'servicesnames': [], 'session_id': 'b6552478f74e655b236cd192c85c9831b69221681ad1f971fb99a1451391b159', 'startingheight': -1, 'subver': '', 'synced_blocks': -1, 'synced_headers': -1, 'timeoffset': 0, 'transport_protocol_type': 'v2', 'version': 0})\r\n```\r\n\r\nCI run: https://cirrus-ci.com/task/4559353030639616?logs=ci#L3823",
      "created_at" : "2024-02-28T10:45:44Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1968712060",
      "id" : 1968712060,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII5851WCl8",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968712060/reactions"
      },
      "updated_at" : "2024-02-28T10:45:44Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968712060",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/1204616?v=4",
         "events_url" : "https://api.github.com/users/m3dwards/events{/privacy}",
         "followers_url" : "https://api.github.com/users/m3dwards/followers",
         "following_url" : "https://api.github.com/users/m3dwards/following{/other_user}",
         "gists_url" : "https://api.github.com/users/m3dwards/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/m3dwards",
         "id" : 1204616,
         "login" : "m3dwards",
         "node_id" : "MDQ6VXNlcjEyMDQ2MTY=",
         "organizations_url" : "https://api.github.com/users/m3dwards/orgs",
         "received_events_url" : "https://api.github.com/users/m3dwards/received_events",
         "repos_url" : "https://api.github.com/users/m3dwards/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/m3dwards/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/m3dwards/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/m3dwards"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "It is in a different line, so it seems different.\r\n\r\nUnrelated, I think it would be useful if the assert helper would sort and pretty-print dictionaries. Otherwise it is hard to spot the difference.",
      "created_at" : "2024-02-28T11:53:02Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1968821663",
      "id" : 1968821663,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII5851WdWf",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 1,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 1,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968821663/reactions"
      },
      "updated_at" : "2024-02-28T11:53:02Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968821663",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=4",
         "events_url" : "https://api.github.com/users/maflcko/events{/privacy}",
         "followers_url" : "https://api.github.com/users/maflcko/followers",
         "following_url" : "https://api.github.com/users/maflcko/following{/other_user}",
         "gists_url" : "https://api.github.com/users/maflcko/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/maflcko",
         "id" : 6399679,
         "login" : "maflcko",
         "node_id" : "MDQ6VXNlcjYzOTk2Nzk=",
         "organizations_url" : "https://api.github.com/users/maflcko/orgs",
         "received_events_url" : "https://api.github.com/users/maflcko/received_events",
         "repos_url" : "https://api.github.com/users/maflcko/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/maflcko/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/maflcko/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/maflcko"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "yes, it's a different issue, most likely related to #29358 - @stratospher and I are looking into it.",
      "created_at" : "2024-02-28T14:39:46Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1969129132",
      "id" : 1969129132,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII5851Xoas",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 1,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 1,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1969129132/reactions"
      },
      "updated_at" : "2024-02-28T14:39:46Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1969129132",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/48763452?v=4",
         "events_url" : "https://api.github.com/users/mzumsande/events{/privacy}",
         "followers_url" : "https://api.github.com/users/mzumsande/followers",
         "following_url" : "https://api.github.com/users/mzumsande/following{/other_user}",
         "gists_url" : "https://api.github.com/users/mzumsande/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/mzumsande",
         "id" : 48763452,
         "login" : "mzumsande",
         "node_id" : "MDQ6VXNlcjQ4NzYzNDUy",
         "organizations_url" : "https://api.github.com/users/mzumsande/orgs",
         "received_events_url" : "https://api.github.com/users/mzumsande/received_events",
         "repos_url" : "https://api.github.com/users/mzumsande/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/mzumsande/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/mzumsande/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/mzumsande"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "Gotcha.\r\n\r\nI have since been able to get what looks like the same issue.\r\n\r\n```shell\r\n node1 2024-02-28T14:39:06.105966Z (mocktime: 2024-02-28T14:39:05Z) [C:\\Users\\Max\\source\\bitcoin\\src\\logging\\timer.h:58] [Log] [addrman] CheckAddrman: completed (0.03ms)\r\n node1 2024-02-28T14:39:06.105977Z (mocktime: 2024-02-28T14:39:05Z) [C:\\Users\\Max\\source\\bitcoin\\src\\logging\\timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started\r\n\r\n test  2024-02-28T14:39:06.106000Z TestFramework (ERROR): Assertion failed\r\n                                   Traceback (most recent call last):\r\n                                     File \"C:\\Users\\Max\\source\\bitcoin\\test\\functional\\test_framework\\test_framework.py\", line 131, in main\r\n                                       self.run_test()\r\n                                     File \"C:\\Users\\Max\\source\\bitcoin/test/functional/rpc_net.py\", line 72, in run_test\r\n                                       self.test_getrawaddrman()\r\n                                     File \"C:\\Users\\Max\\source\\bitcoin/test/functional/rpc_net.py\", line 512, in test_getrawaddrman\r\n                                       check_getrawaddrman_entries(expected)\r\n                                     File \"C:\\Users\\Max\\source\\bitcoin/test/functional/rpc_net.py\", line 430, in check_getrawaddrman_entries\r\n                                       assert_equal(len(getrawaddrman[table_name]), len(table_info[\"entries\"]))\r\n                                     File \"C:\\Users\\Max\\source\\bitcoin\\test\\functional\\test_framework\\util.py\", line 57, in assert_equal\r\n                                       raise AssertionError(\"not(%s)\" % \" == \".join(str(arg) for arg in (thing1, thing2) + args))\r\n                                   AssertionError: not(3 == 2)\r\n node1 2024-02-28T14:39:06.106021Z (mocktime: 2024-02-28T14:39:05Z) [C:\\Users\\Max\\source\\bitcoin\\src\\logging\\timer.h:58] [Log] [addrman] CheckAddrman: completed (0.04ms)\r\n node1 2024-02-28T14:39:06.106033Z (mocktime: 2024-02-28T14:39:05Z) [C:\\Users\\Max\\source\\bitcoin\\src\\logging\\timer.h:58] [Log] [addrman] CheckAddrman: new 3, tried 1, total 4 started\r\n```",
      "created_at" : "2024-02-28T14:47:40Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1969145053",
      "id" : 1969145053,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII5851XsTd",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1969145053/reactions"
      },
      "updated_at" : "2024-02-28T14:47:40Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1969145053",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/1204616?v=4",
         "events_url" : "https://api.github.com/users/m3dwards/events{/privacy}",
         "followers_url" : "https://api.github.com/users/m3dwards/followers",
         "following_url" : "https://api.github.com/users/m3dwards/following{/other_user}",
         "gists_url" : "https://api.github.com/users/m3dwards/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/m3dwards",
         "id" : 1204616,
         "login" : "m3dwards",
         "node_id" : "MDQ6VXNlcjEyMDQ2MTY=",
         "organizations_url" : "https://api.github.com/users/m3dwards/orgs",
         "received_events_url" : "https://api.github.com/users/m3dwards/received_events",
         "repos_url" : "https://api.github.com/users/m3dwards/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/m3dwards/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/m3dwards/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/m3dwards"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "This can be closed after #29007 merge (#28998 could still make sense, but the CI failure cannot happen anymore).",
      "created_at" : "2024-03-11T19:10:02Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/28964#issuecomment-1989234446",
      "id" : 1989234446,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/28964",
      "node_id" : "IC_kwDOABII5852kU8O",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1989234446/reactions"
      },
      "updated_at" : "2024-03-11T19:10:02Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1989234446",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/48763452?v=4",
         "events_url" : "https://api.github.com/users/mzumsande/events{/privacy}",
         "followers_url" : "https://api.github.com/users/mzumsande/followers",
         "following_url" : "https://api.github.com/users/mzumsande/following{/other_user}",
         "gists_url" : "https://api.github.com/users/mzumsande/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/mzumsande",
         "id" : 48763452,
         "login" : "mzumsande",
         "node_id" : "MDQ6VXNlcjQ4NzYzNDUy",
         "organizations_url" : "https://api.github.com/users/mzumsande/orgs",
         "received_events_url" : "https://api.github.com/users/mzumsande/received_events",
         "repos_url" : "https://api.github.com/users/mzumsande/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/mzumsande/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/mzumsande/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/mzumsande"
      }
   }
]
